File: MyLanguageWeb/src/main/resources/static/js/classes/dto/entity/word/types/words_statistic_response_dto.js
import {
    WordStatusResponseDTO
} from "../../word_status/response/word_status_response_dto.js";

export class WordsStatisticResponseDTO {
    #numberOfWords;
    #wordStatus;

    constructor(json) {
        if (json) {
            this.#numberOfWords = json["number_of_words"];

            let wordStatus = json["word_status"];
            if (wordStatus) {
                this.#wordStatus = new WordStatusResponseDTO(wordStatus);
            }
        }
    }

    getNumberOfWords() {
        return this.#numberOfWords;
    }

    getWordStatus() {
        return this.#wordStatus;
    }
}

File: MyLanguageWeb/src/main/java/ru/herooo/mylanguageweb/dto/entity/word/types/statistic/WordsStatisticResponseDTO.java
package ru.herooo.mylanguageweb.dto.entity.word.types.statistic;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import ru.herooo.mylanguageweb.controllers.json.l.LongDeserializer;
import ru.herooo.mylanguageweb.controllers.json.l.LongSerializer;
import ru.herooo.mylanguageweb.dto.entity.wordstatus.response.WordStatusResponseDTO;

public class WordsStatisticResponseDTO {
    @JsonSerialize(using = LongSerializer.class)
    @JsonDeserialize(using = LongDeserializer.class)
    @JsonProperty("number_of_words")
    private long numberOfWords;

    @JsonProperty("word_status")
    private WordStatusResponseDTO wordStatus;

    public long getNumberOfWords() {
        return numberOfWords;
    }

    public void setNumberOfWords(long numberOfWords) {
        this.numberOfWords = numberOfWords;
    }

    public WordStatusResponseDTO getWordStatus() {
        return wordStatus;
    }

    public void setWordStatus(WordStatusResponseDTO wordStatus) {
        this.wordStatus = wordStatus;
    }
}
